src/main/java/org/assertj/core/api/ComparatorFactory.java

```diff
@@ -12,21 +12,58 @@
  */
 package org.assertj.core.api;
 
-import static java.lang.Math.abs;
-
+import java.math.BigDecimal;
 import java.util.Comparator;
+import java.util.Objects;
 
 public class ComparatorFactory {
 
   public static final ComparatorFactory INSTANCE = new ComparatorFactory();
 
+  public static <T extends Number & Comparable<T>> boolean isNanOrInfinity(T number){
+    // avoid null argument
+    if(Objects.requireNonNull(number) instanceof Float){
+      return ((Float) number).isNaN() || ((Float) number).isInfinite();
+    }else if(Objects.requireNonNull(number) instanceof Double){
+      return ((Double)number).isNaN() || ((Double)number).isInfinite();
+    }else {
+      return false;
+    }
```

Comment:
@YeeTone please address this comment

Reply:
@YeeTone please address this comment and the following I already made in a previous review: you don't need `Objects.requireNonNull(number)` since instanceof returns false for null

Reply:
Solved!